feature/topic-operator: Add SASL, SCRAM and Public CA connectivity options (#5201)
strimzi/strimzi-kafka-operator: documentation/modules/deploying/proc-deploy-topic-operator-standalone.adoc

Default `/strimzi/topics`.
.. `STRIMZI_TLS_ENABLED` to enable TLS support for encrypting the communication with Kafka brokers.
Default `true`.

> .. `STRIMZI_PUBLIC_CA` to skip generating trust store certificates if TLS is enabled.

Reviewer: Let's explain the effect that this will have on the operator.

Default `false`.

> .. `STRIMZI_TLS_AUTH_ENABLED` to skip generating key store certificates if TLS is enabled.

Reviewer: Again, what does that mean for the operator?

<9> Enables TLS support for encrypted communication with the Kafka brokers.
<10> (Optional) The Java options used by the JVM running the Topic Operator.
<11> (Optional) The debugging (`-D`) options set for the Topic Operator.

> <12> (Optional) Skip generating trust store certificates if TLS is enabled. If this setting is enabled then the brokers have to use a public trusted certificate authority for their TLS certificates.

Reviewer 2: ```suggestion
<12> (Optional) Skips the generation of trust store certificates if TLS is enabled. If this environment variable is enabled, the brokers must use a public trusted certificate authority for their TLS certificates.
```

> The default is "false".

Reviewer 2: ```suggestion
The default is `false`.
```

> <13> (Optional) Generate key store certificates for mutual TLS authentication. Setting this to false will disable client authentication with TLS to the Kafka brokers.
> The default is "true".

Reviewer 2: ```suggestion
<13> (Optional) Generates key store certificates for mutual TLS authentication. Setting this to `false` disables client authentication with TLS to the Kafka brokers.
The default is `true`.
```

> <14> (Optional) Enable SASL support for client authentication when connecting to Kafka brokers.
> The default is "false".

Reviewer 2: ```suggestion
<14> (Optional) Enables SASL support for client authentication when connecting to Kafka brokers.
The default is `false`.
```

<15> (Optional) The SASL username for client authentication.
Mandatory only if SASL is enabled through `STRIMZI_SASL_ENABLED`.
<16> (Optional) The SASL password for client authentication.
Mandatory only if SASL is enabled through `STRIMZI_SASL_ENABLED`.
<17> (Optional) The SASL mechanism for client authentication.
Mandatory only if SASL is enabled through `STRIMZI_SASL_ENABLED`.

> The value can be set to: `plain`, `scram-sha-256` and `scram-sha-512`.

Reviewer 2: ```suggestion
You can set the value to `plain`, `scram-sha-256`, or `scram-sha-512`.
```

<18> (Optional) The security protocol used for communication with Kafka brokers.
The default value is "PLAINTEXT".

> The value can be set to: `PLAINTEXT`, `SSL`, `SASL_PLAINTEXT` and `SASL_SSL`.

Reviewer 2: ```suggestion
You can set the value to `PLAINTEXT`, `SSL`, `SASL_PLAINTEXT`, or `SASL_SSL`.
```

> . If you enabled TLS with the `STRIMZI_TLS_ENABLED` environment variable, specify the keystore and truststore used to authenticate connection to the Kafka cluster.

Reviewer 2: Do we need to update this to mention `STRIMZI_PUBLIC_CA`?  Would we better moving the new optional connectivity environment variables into another separate step so we can give the variables a bit of context? _If you want to connect...use these variables..._

The default is `false`.
<13> (Optional) Generates key store certificates for mutual TLS authentication. Setting this to `false` disables client authentication with TLS to the Kafka brokers.
The default is `true`.
<14> (Optional) Enables SASL support for client authentication when connecting to Kafka brokers.
The default is `false`.
<15> (Optional) The SASL username for client authentication.
Mandatory only if SASL is enabled through `STRIMZI_SASL_ENABLED`.
<16> (Optional) The SASL password for client authentication.
Mandatory only if SASL is enabled through `STRIMZI_SASL_ENABLED`.
<17> (Optional) The SASL mechanism for client authentication.
Mandatory only if SASL is enabled through `STRIMZI_SASL_ENABLED`.
You can set the value to `plain`, `scram-sha-256`, or `scram-sha-512`.
<18> (Optional) The security protocol used for communication with Kafka brokers.
The default value is "PLAINTEXT".
You can set the value to `PLAINTEXT`, `SSL`, `SASL_PLAINTEXT`, or `SASL_SSL`.

. If you enabled TLS with the `STRIMZI_TLS_ENABLED` environment variable, specify the keystore and truststore used to authenticate connection to the Kafka cluster.

> . If you want to connect to Kafka brokers which are using certificates from public CAs as for example AWS MSK then you'll need to set `STRIMZI_PUBLIC_CA` to `true`.

Reviewer 2: ```suggestion
. If you want to connect to Kafka brokers that are using certificates from public CAs, set `STRIMZI_PUBLIC_CA` to `true`. Set this property to `true`, for example, if you are using Kafka on Amazon Web Services (AWS).
```

+

> .Example TLS configuration

Reviewer 2: Should step 2 and 3 swap places so that the example truststore confif sits with _If you enabled TLS with the STRIMZI_TLS_ENABLED environment variable,..._

Reviewer 2: Could you remove the quote from the example config?
` # ..."` - > `# ...`
